src/wagtailtrans/views/language.py

```diff
@@ -0,0 +1,23 @@
+from __future__ import absolute_import, unicode_literals
+
+from wagtail.contrib.modeladmin.views import DeleteView
+
+
+class LanguageDeleteView(DeleteView):
+    """
+    Custom Delete View Class for `Language` ModelAdmin.
+    """
+    def post(self, request, *args, **kwargs):
+        """Overriding Default post method because we need to delete related
+        translatable pages (if there is any) before we can delete a Language
+        which are references through protected foreign key.
+
+        Also we'll not allow to delete the default language.
+        """
+        if self.instance.is_default:
+            raise Exception("Can't delete a default language")
```

Comment:
I would raise a 403 here. instead of 500 error.
Also, maybe it might be nice to check if the language is default before actually doing the post.

Reply:
1. Yes agreed. I'll throw a 403 instead.

2. Well, we don't have to, because in a normal template flow you'll never be in the `post` endpoint. we have different template so user will not be able to make a `post` request via admin/dashboard in case of a default language. But this check ensure that even if it does able to make the request we won't be processing this.